### PR TITLE
Add a utility class for parallel reading from ArrowStreams

### PR DESCRIPTION
**Pull request by emkornfield**
I'll incorporate this into the actual Arrow flow in a separate PR.

### CODE REVIEW

**Review by davidrabinowitz — CHANGES_REQUESTED** [review]

File: connector/src/main/java/com/google/cloud/bigquery/connector/common/ParallelArrowReader.java [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+          queue.put(
+              executor.submit(
+                  () -> {
+                    try {
+                      hasData[idx].set(reader.loadNextBatch());
+                    } catch (IOException e) {
+                      readException = e;
+                      return null;
+                    } catch (Exception e) {
+                      readException = new IOException(e);
```
[the remark this answers is not in the document]
Comment (emkornfield) [reply connector/src/main/java/com/google/cloud/bigquery/connector/common/ParallelArrowReader.java]: done.

**Review by davidrabinowitz — APPROVED** [review]

**Pull request merged**